plugins/morph_resultsdb.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/python
+import argparse
+import logging
+import logging.config
+import requests
+import json
+import time
+
+from lib.logging_conf import setup_logging
+
+
+class ResultsDBApiException(Exception):
+    pass
+
+
+class ResultsDBApi(object):
+    """
+    Class to communicate and process data with resultsDB
+    """
+    TIMEOUT_LIMIT = 7200  # Wait 2 hours maximally
+    RESULTSDB_LIMITER = 10
+    MINUTE = 60  # 60 seconds
+
+    def __init__(self, job_names, component_nvr, test_tier, resultsdb_api_url):
+        self.resultsdb_api_url = resultsdb_api_url
+        self.job_names = job_names
+        self.job_names_result = {}
+        self.tier_tag = True
+        self.url_options = {'CI_tier': test_tier, 'item': component_nvr}
+
+    def get_resultsdb_data(self):
+        """
+        Method get_resultsdb_data it's the main function from which data are received
+
+        :returns -- dictionary where keys are job names and their values are list of queried data
+        """
+        if self.job_names:
+            return self.get_job_names_result()
+        else:
+            queried_data = self.get_jobs_by_nvr_and_tier(self.RESULTSDB_LIMITER)
+            self.job_names_result = self.setup_output_data(queried_data)
+            return self.job_names_result
+
+    @staticmethod
+    def query_resultsdb(url, url_options=dict):
+        """
+        This method queries resultsDB with given url and url_option variable
+
+        :param url -- resultdb api url
+        :param url_options -- dictionary of wanted options
+
+        :returns -- Queried data
+        """
+        logging.debug("Running resultsbd API query with this url: {0} and options {1}".format(url, url_options))
+        response = requests.get(url, params=url_options)
+        if response.status_code >= 300:
+            logging.error("ERROR: Unable to access resultsdb site.")
+            raise ResultsDBApiException("ERROR: Unable to access resultsdb site.")
```

Comment:
Can we retry this for 3 times with sleep of 120 seconds before raising ResultsDBApiException ?